File: LUA/10-binary-numbers.lua
local n = io.read("*n")
maxConsecutives = 0
counter = 0

while n > 0 do
	rest = n % 2

	if rest == 1 then
		counter = counter + 1
	else
		if counter > maxConsecutives then
			maxConsecutives = counter
		end
		counter = 0
	end

	n = (n-rest)/2
end

if counter > maxConsecutives then
	maxConsecutives = counter
end

io.write(maxConsecutives)

File: JAVA/10-binary-numbers.java
import java.util.Scanner;

class BinaryNumbers
{
  private static final Scanner scanner = new Scanner(System.in);
  
  public static void main(String[] args)
  {
    scanner.skip("(\r\n|[\n\r\u2028\u2029\u0085])?");
    int    n               = scanner.nextInt();
    char[] binary          = Integer.toBinaryString(n).toCharArray();
    int    maxConsecutives = 0;
    int    counter         = 0;
    
    for (int i = 0; i < binary.length; i++)
      if (binary[i] == '1')
        counter++;
      else
      {
        //end if
        if (counter > maxConsecutives)
          maxConsecutives = counter;
        counter = 0;
      }// end if-else
    
    if (counter > maxConsecutives)
      maxConsecutives = counter;
    
    System.out.println(maxConsecutives);
    scanner.close();
  }// end main
}// end BinaryNumbers - class